infracloudio/botkube
# Implement E2E tests against BotKube deployed on K8s and real Slack API

##### ISSUE TYPE

 - Feature Pull Request

##### SUMMARY

- Implement E2E tests against BotKube deployed on K8s and real Slack API
- Allow building Docker image
- Ensure output from FilterEngine and Executor are predictable (same order every map iteration)
- Change missing TLS secret from Recommendation to Warning
- Add happy path unit test for Ingress validation filter (to make sure we will be able to remove current E2E tests with fake Slack and K8s in near future as the coverage is equal or higher)

This PR will be followed-up by a dedicated PR which run these tests on CI (with `helm test`).

See https://github.com/infracloudio/botkube/issues/615

##### TESTING

1. Check out this PR
1. See the testing instruction in the `./tests` package.
